src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/PerformanceCounterLib.cs

```diff
@@ -50,11 +50,13 @@ internal class PerformanceCounterLib
         private string _machineName;
         private string _perfLcid;
 
-        private Hashtable _customCategoryTable;
+
         private static volatile Hashtable s_libraryTable;
+        private Hashtable _customCategoryTable;
         private Hashtable _categoryTable;
         private Hashtable _nameTable;
         private Hashtable _helpTable;
+        private readonly object _customCategoryTableLock = new Object();
```

Comment:
I would recommend only using InternalSyncObject.  Introducing a new lock here can cause deadlocks if we don't get it right.  I've seen that sort of thing in this area before and we've removed locks for this reason.

Reply:
Understood

Reply:
There is nothing fundamentally wrong with this.  For me the question is whether or not the number of instances and actions upon them in a multi-threaded fashion will result in significant performance degradation if we use a single lock.  My suspicion is that this won't be an issue because this is all about creation of counters and not use of counters.  I would be much more concerned if we did something like this when reading counters.

Reply:
i agree, seems ok, checked with call hierarchy tool, call only in init paths(_initialized). 
![callhierarchy](https://user-images.githubusercontent.com/7894084/36627815-f16a3ce4-1948-11e8-965c-4bf4bac77e1b.png)